src/main/java/programminglife/gui/controller/GuiController.java

```diff
@@ -201,23 +203,25 @@ private void initMenubar() {
             }
             try {
                 file = fileChooser.showOpenDialog(ProgrammingLife.getStage());
-                this.openFile(file);
-                try (BufferedWriter fw = new BufferedWriter(new FileWriter(recentFile, true))) {
-                    if (!recentItems.contains(file.getAbsolutePath())) {
-                        fw.write(file.getAbsolutePath() + System.getProperty("line.separator"));
-                        fw.flush();
-                        fw.close();
-                        initRecent();
+                if (file != null) {
+                    this.openFile(file);
+                    try (BufferedWriter fw = new BufferedWriter(new FileWriter(recentFile, true))) {
```

Comment:
This was already like this, I just added `if (file != null) {...}` to prevent `NullPointerException` when canceling the `FileOpenDialog`. But, I can rename it for clarity.